tools/buildscripts/release.py

```diff
@@ -0,0 +1,338 @@
+                sys.exit(1)
+
+        for mod_name, mod_info in self.cpp_mods.items():
+            if self.is_dev_release or mod_info['include_in_release']:
+                if mod_info['create_config']:
+                    os.makedirs(os.path.join(self.ue4ss_dir, 'Mods', mod_name, 'config'), exist_ok=True)
+
+    def modify_mods_txt(self):
+        mods_to_disable_in_release = self.mods_to_disable_in_release.copy()
+        for mod_name, mod_info in self.cpp_mods.items():
+            if not mod_info['include_in_release']:
+                mods_to_disable_in_release[mod_name] = 0
+
+        mods_path = os.path.join(self.ue4ss_dir, 'Mods/mods.txt')
+        with open(mods_path, mode='r', encoding='utf-8-sig') as file:
+            content = file.read()
+
+        if self.cpp_mods:
+            content = '\n'.join([f'{mod} : 1' for mod in self.cpp_mods]) + '\n' + content
+
+        if not self.is_dev_release:
+            for key, value in mods_to_disable_in_release.items():
+                pattern = rf'(^{key}\s*:).*?$'
+                content = re.sub(pattern, rf'\1 {value}', content, flags=re.MULTILINE)
+
+        with open(mods_path, mode='w', encoding='utf-8-sig') as file:
+            file.write(content)
+
+    def modify_mods_json(self):
+        mods_path = os.path.join(self.ue4ss_dir, 'Mods/mods.json')
+        with open(mods_path, mode='r', encoding='utf-8-sig') as file:
+            content = json.load(file)
+
+        if self.cpp_mods:
+            for mod_name, mod_info in self.cpp_mods.items():
+                if mod_name not in [mod['mod_name'] for mod in content]:
+                    if not self.is_dev_release:
+                        content.append({'mod_name': mod_name, 'mod_enabled': mod_info['include_in_release']})
+                    else:
+                        content.append({'mod_name': mod_name, 'mod_enabled': True})
+
+        for mod in content:
+            if mod['mod_name'] in self.mods_to_disable_in_release:
+                mod['mod_enabled'] = bool(self.mods_to_disable_in_release[mod['mod_name']])
+
+        with open(mods_path, mode='w', encoding='utf-8-sig') as file:
+            json.dump(content, file, indent=4)
+
+    def copy_executables(self, dwmapi_dll_path, ue4ss_dll_path, ue4ss_pdb_path):
+        shutil.copy(dwmapi_dll_path, self.staging_dir)
+        shutil.copy(ue4ss_dll_path, self.ue4ss_dir)
+        if self.is_dev_release: shutil.copy(ue4ss_pdb_path, self.ue4ss_dir)
+
+    def copy_docs(self):
+        if self.is_dev_release and os.path.exists('docs'):
+            shutil.copytree('docs', os.path.join(self.ue4ss_dir, 'Docs'))
+
+    def modify_changelog(self):
+        changelog_path = self.ue4ss_dir + '/Changelog.md'
+        if os.path.exists(changelog_path): ready_changelog_for_release(changelog_path)
+
+    def package_release(self):
+        version = subprocess.check_output(['git', 'describe', '--tags']).decode('utf-8').strip()
```

Comment:
Nitpick: It might be worth catching the error here to make it a little easier for people investigating failed releases? The current output (when erroring) on my machine is 

```
fatal: No names found, cannot describe anything.
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Python39\lib\subprocess.py", line 424, in check_output
    return run(*popenargs, stdout=PIPE, timeout=timeout, check=True,
  File "C:\Python39\lib\subprocess.py", line 528, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['git', 'describe', '--tags']' returned non-zero exit status 128.
```

Pruning and shallow cloning in the checkout step would potentially cause the git describe --tags to not work as expected and so we probably should handle it gracefully downstream of that step?